### ISSUE TITLE
[C++20] Clang doesn't consider aggregate type as trivially constructible

### ISSUE THREAD

**Issue by danakj** [llvm/llvm-project#55800]
This is related to https://github.com/llvm/llvm-project/issues/55799. This works in GCC and MSVC when building C++20:

```cpp
#include <type_traits>

struct S {
    int i;
};

static_assert(std::is_trivially_constructible_v<S, int>, "");
```

GCC: https://godbolt.org/z/r1xdzP4oo
MSVC: https://godbolt.org/z/c7s7ezE16
Clang: https://godbolt.org/z/cGbqro5jE

```
<source>:7:1: error: static_assert failed due to requirement 'std::is_trivially_constructible_v<S, int>' ""
static_assert(std::is_trivially_constructible_v<S, int>, "");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by alanzhao1** [comment]
This is currently blocked on #55799 

There is a [candidate patch](https://reviews.llvm.org/D129531) for #55799 that is required to resolve this bug as that patch implements the constructor syntax this bug refers to, but currently that patch doesn't work.

**Comment by 0x59616e** [comment]
> This is currently blocked on #55799
> 
> There is a [candidate patch](https://reviews.llvm.org/D129531) for #55799 that is required to resolve this bug as that patch implements the constructor syntax this bug refers to, but currently that patch doesn't work.

There are some bugs in this patch. I'm starting to fix this.

**Comment by alanzhao1** [comment]
Resolved now that #54040 has been resolved by 40c5215
